src/blocks/content-helper/content-helper.tsx

```diff
@@ -12,19 +12,20 @@ import { registerPlugin } from '@wordpress/plugins';
 import PerformanceDetails from './performance-details/component';
 import RelatedTopPostList from './related-top-posts/component-list';
 import LeafIcon from '../shared/components/leaf-icon';
+import { ElementOrEmptyCredentialsMessage } from './content-helper-error';
 
 const BLOCK_PLUGIN_ID = 'wp-parsely-block-editor-sidebar';
 
 const renderSidebar = () => (
 	<PluginSidebar icon={ <LeafIcon /> } name="wp-parsely-content-helper" className="wp-parsely-content-helper" title={ __( 'Parse.ly Editor Sidebar', 'wp-parsely' ) }>
 		<Panel>
 			<PanelBody title={ __( 'Performance Details', 'wp-parsely' ) } initialOpen={ true }>
-				<PerformanceDetails />
+				{ ElementOrEmptyCredentialsMessage( <PerformanceDetails /> ) }
```

Comment:
Is it better if we implement it this way?

```jsx
<VerifyCredentials>
    <PerformanceDetails />
</VerifyCredentials>
```

Here `VerifyCredentials` will show the component passed to it via `props.children` if credentials are good else show the empty credentials message.

Reply:
Yes! Implemented in https://github.com/Parsely/wp-parsely/pull/1588/commits/a8901418d6859995bc4df3c84ce1b77bee1bd72b.